source/extensions/listener_managers/listener_manager/listener_manager_impl.cc

```diff
@@ -350,6 +350,8 @@ ListenerManagerImpl::ListenerManagerImpl(Instance& server,
                                          bool enable_dispatcher_stats,
                                          Quic::QuicStatNames& quic_stat_names)
     : server_(server), factory_(std::move(factory)),
+      filter_config_provider_manager_(
+          std::make_shared<Filter::HttpFilterConfigProviderManagerImpl>()),
```

Comment:
This is incorrect - it should be a singleton provider, that is shared across the listeners. It looks like you create one per listener here and in the server factory context.